graph/node/node.go

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+// Node tracks the metadata associated with a node in the graph
+type Node struct {
+	ID string
+
+	value interface{}
+}
+
+// New creates a new node
+func New(id string, value interface{}) *Node {
```

Comment:
yeah, we need to not lose accumulated metadata… `SetValue` works but it means we're mutating the nodes. We could copy them, of course, but that essentially means throwing away an entire graph at every stage. I'm not sure what the right solution is. Since our processing only goes forward and never forks, mutating might make sense. What do you think?